listenbrainz_spark/stats/user/utils.py

```diff
@@ -25,30 +23,23 @@ def get_artists(table):
                 }
     """
     t0 = time.time()
-    result = run_query("""
-            SELECT user_name
-                 , artist_name
-                 , artist_msid
-                 , artist_mbids
-                 , count(artist_name) as cnt
-              FROM %s
-          GROUP BY user_name, artist_name, artist_msid, artist_mbids
-          ORDER BY cnt DESC
-        """ % (table))
-    rows = result.collect()
+    df = df.select("user_name","artist_name","artist_msid","artist_mbids")
+    df = df.groupBy("user_name","artist_name","artist_msid","artist_mbids").count()
+    df = df.sort(df['count'].desc())
+    rows = df.collect() 
     artists = defaultdict(list)
     for row in rows:
         artists[row.user_name].append({
             'artist_name': row.artist_name,
             'artist_msid': row.artist_msid,
             'artist_mbids': row.artist_mbids,
-            'listen_count': row.cnt,
+            'listen_count': row['count']
```

Comment:
I don't clearly understand this. imo, count is a member of row object which should be referenced using dot operator.